src/backend/distributed/master/master_delete_protocol.c

```diff
@@ -443,6 +476,53 @@ DropShards(Oid relationId, char *schemaName, char *relationName,
 }
 
 
+/*
+ * DropTaskList returns a list of tasks to execute a DROP command on shard
+ * placements of distributed table. This is handled separately from other
+ * DDL commands because we handle it via the DROP trigger, which is called
+ * whenever a truncate cascades.
+ */
+static List *
+DropTaskList(Oid relationId, char *schemaName, char *relationName,
+			 List *deletableShardIntervalList)
+{
+	/* resulting task list */
+	List *taskList = NIL;
+
+	ShardInterval *shardInterval = NULL;
+	foreach_ptr(shardInterval, deletableShardIntervalList)
+	{
+		Assert(shardInterval->relationId == relationId);
+
+		uint64 shardId = shardInterval->shardId;
+		char storageType = shardInterval->storageType;
+
+		char *shardRelationName = pstrdup(relationName);
+
+		/* build shard relation name */
+		AppendShardIdToName(&shardRelationName, shardId);
+
+		char *dropShardPlacementCommand =
+			CreateDropShardPlacementCommand(schemaName, shardRelationName,
+											storageType);
+
+		Task *task = CitusMakeNode(Task);
+		task->jobId = INVALID_JOB_ID;
+		task->taskId = INVALID_TASK_ID;
+		task->taskType = DDL_TASK;
+		SetTaskQueryString(task, dropShardPlacementCommand);
+		task->dependentTaskList = NULL;
+		task->replicationModel = REPLICATION_MODEL_INVALID;
+		task->anchorShardId = shardId;
+		task->taskPlacementList = ShardPlacementList(shardId);
```

Comment:
This should be `ActiveShardPlacementList` ?

 And, I think we should set `task->taskId`, see how VACUUM does it.

In general, I'd be much happier if we can have a separate PR for `DropTaskList ` on master

Reply:
I can move things to another pr.

> And, I think we should set task->taskId, see how VACUUM does it.

as we execute drop tasks in a seperate logic (out of the adaptive executor function calls), we do not need such taskId `fields` etc.

> This should be ActiveShardPlacementList ?

I think we want to delete all shard placement rows whether or not they are active etc. this is why we operate on all shard placements of it

Reply:
edit: actually the changes I performed on `DropShards` are related with local drop execution. Hence them moving to another pr wouldn't make much sense I guess ?

Reply:
I think it still makes sense to have a small PR for `DropTaskList ` as it is completely different than this PR

Reply:
> I think we want to delete all shard placement rows whether or not they are active etc. this is why we operate on all shard placements of it

Oh, yes that seems correct.

Reply:
> I think it still makes sense to have a small PR for DropTaskList as it is completely different than this PR

opened another pr at https://github.com/citusdata/citus/pull/3603